browser/brave_webcompat_reporter/webcompat_reporter_service.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_webcompat_reporter/webcompat_reporter_service.h"
+
+#include <string>
+
+#include "net/base/load_flags.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "base/task/post_task.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "chrome/browser/browser_process.h"
+#include "base/json/json_writer.h"
+#include "base/environment.h"
+#include "brave/components/brave_referrals/browser/brave_referrals_service.h"
+#include "brave/common/network_constants.h"
+
+namespace brave {
+
+BraveWebcompatReporterService::BraveWebcompatReporterService() {}
+
+BraveWebcompatReporterService::~BraveWebcompatReporterService() {}
+
+void BraveWebcompatReporterService::SubmitReport(std::string report_domain) {
+  std::string api_key = brave::GetAPIKey();
+
+  std::unique_ptr<base::Environment> env(base::Environment::Create());
+  std::string upload_url;
+  env->GetVar("BRAVE_WEBCOMPAT_REPORT_ENDPOINT", &upload_url);
```

Comment:
I don't think we should be using an env var for this, it should be configured through gn and if we need an override for it we can use a command line switch

Reply:
I took the same approach as the [referral server URL](https://github.com/brave/brave-core/blob/c8083458ecf718f35ad6f2f80a917f51d1c95264/components/brave_referrals/browser/brave_referrals_service.cc#L85). Do you have any examples of strings set this way I could refer to?

Reply:
`safebrowsing_api_endpoint`

Reply:
I've updated the description to reference https://github.com/brave/brave-browser/pull/6550.